Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -960,6 +1013,60 @@ static String mutationObserverNotificationString()
     }
 }
 
+#if ENABLE(UNIFIED_PDF_PAINT_SELECTIONS)
+void UnifiedPDFPlugin::paintPDFSelection(GraphicsContext& context, const FloatRect& clipRect)
+{
+    if (!m_currentSelection || [m_currentSelection isEmpty])
+        return;
+
+    bool isVisibleAndActive = false;
+    if (RefPtr page = this->page())
+        isVisibleAndActive = page->isVisibleAndActive();
+
+    auto selectionColor = [renderer = m_element->renderer(), isVisibleAndActive] {
+        auto& renderTheme = renderer->theme();
+        auto styleColorOptions = renderer->styleColorOptions();
+        auto selectionColorWithAlpha = isVisibleAndActive ? renderTheme.activeSelectionBackgroundColor(styleColorOptions) : renderTheme.inactiveSelectionBackgroundColor(styleColorOptions);
+        // We want to paint opaquely and composite via the selection layer itself, which is configured with a "multiply" blend mode.
+        return selectionColorWithAlpha.opaqueColor();
+    }();
+
+    auto pageCoverage = pageCoverageForRect(clipRect);
+    auto documentScale = pageCoverage.pdfDocumentScale;
+    for (auto& pageInfo : pageCoverage.pages) {
+        auto page = m_documentLayout.pageAtIndex(pageInfo.pageIndex);
+        if (!page || !shouldDisplayPage(pageInfo.pageIndex))
+            continue;
+
+        auto pageDestinationRect = pageInfo.pageBounds;
+
+        GraphicsContextStateSaver pageStateSaver { context };
+        context.scale(documentScale);
+        context.clip(pageDestinationRect);
+        // Translate the context to the bottom of pageBounds and flip, so that PDFKit operates
+        // from this page's drawing origin.
+        context.translate(pageDestinationRect.minXMaxYCorner());
+        context.scale({ 1, -1 });
+
+        auto pageGeometry = m_documentLayout.geometryForPage(page);
+        auto transformForBox = m_documentLayout.toPageTransform(*pageGeometry).inverse().value_or(AffineTransform { });
+        context.concatCTM(transformForBox);
+
+        if ([m_currentSelection respondsToSelector:@selector(enumerateRectsAndTransformsForPage:usingBlock:)]) {
+            [protectedCurrentSelection() enumerateRectsAndTransformsForPage:page.get() usingBlock:[&context, &selectionColor](CGRect cgRect, CGAffineTransform cgTransform) mutable {
+                // FIXME: Perf optimization -- consider coalescing rects by transform.
+                GraphicsContextStateSaver individualRectTransformPairStateSaver { context };
+
+                if (AffineTransform transform { cgTransform }; !transform.isIdentity())
+                    context.concatCTM(transform);
+
+                context.fillRect({ cgRect }, selectionColor, CompositeOperator::SourceOver);
```

Comment:
you don't need to say SourceOver here

Reply:
Dropped in https://github.com/WebKit/WebKit/commit/ab434ecd8a237eb5256d62fbc8744be2db5813fc